### PR TITLE
Add fetchMyTrades, fetchTransactions, fetchDeposits, fetchWithdrawals to bittrex

### PR DESCRIPTION
**Pull request by buddylindsey**
Here is another. This built locally lets hope automation works.

I am calling `fetchDeposits` and `fetchWithdrawals` from `fetchTransactions` since fiat currency transfers filter through those endpoints.

In `parseTransaction` I have a note that sometimes bittrex returns back in deposits their `1.0` api data structure. I have it coded out to give priority to the 1.1 data structure and then override for 1.0 when necessary. I am working with bittrex for them to figure out why it is happening.

Also in my conversation with bittrex they intend to release a new version of their API by the end of the year. So this will probably need an overhaul at some point.

### CODE REVIEW

File: js/bittrex.js [hunk 1]
```diff
@@ -504,6 +527,120 @@ module.exports = class bittrex extends Exchange {
...
 
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        let market = undefined;
+        await this.loadMarkets ();
+        let request = {};
+        if (symbol !== undefined)
+            request['symbol'] = this.market (symbol)['id'];
+        if (limit !== undefined)
+            request['count'] = limit;
+        let response = await this.accountGetOrderhistory (this.extend (request, params));
```
Comment (kroitor) [inline js/bittrex.js]: The same is already done in fetchClosedOrders for Bittrex, so, I guess, they don't have an endpoint for fetchMyTrades? This was the reasoning behind setting has['fetchMyTrades'] to false. We actually had it implemented in the same way previously, but, in reality, orders don't always match one-to-one, therefore each order contains one or more trades. Of course, Bittrex hides it from the user with their API, and, ultimately, the users sees those fills as one-to-one. But we think it's better to have as less code in the lib as possible, that is, if the users wants to derive trades from orders, then it is probably better to do that in userland? I'm interested in what you think about it )

Comment (buddylindsey) [reply js/bittrex.js]: I was not aware that took place.

So after doing some digging into data, apis, and talking it over with a colleague. We realized that yes they do not have a fetch my trades endpoint. In fact we could find no way to tie individual trades with orders, in any retrospective way. Also the bittrex API is quite terrible in a lot of ways in getting useful detailed data efficiently. 

With that in mind I am going to rip out the `fetchMyTrades` work, it just isn't a good fit. Maybe with the new API they will add it, but after digging through the UI and csv exports. I doubt they will.

Comment (kroitor) [reply js/bittrex.js]: Yeah, no way to paginate through our own history with Bittrex is a bit dissapointing... Not mentioning the other issues with it... But I hope to see at least some improvements from them by the end of the year )

**Comment by buddylindsey** [comment]
Okay `fetchMyTrades` is removed from PR.

**Pull request merged**